edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/SubscriptionProcessor.cs

```diff
@@ -113,21 +113,24 @@ async Task ProcessSubscription(string id, Option<ICloudProxy> cloudProxy, Device
         async void CloudConnectivityEstablished(object sender, EventArgs eventArgs)
         {
             Events.DeviceConnectedProcessingSubscriptions();
-            try
+            async Task ProcessIdentity(IIdentity identity)
             {
-                IEnumerable<IIdentity> connectedClients = this.ConnectionManager.GetConnectedClients().ToList();
-                foreach (IIdentity identity in connectedClients)
+                try
                 {
-                    try
-                    {
-                        Events.ProcessingSubscriptionsOnDeviceConnected(identity);
-                        await this.ProcessExistingSubscriptions(identity.Id);
-                    }
-                    catch (Exception e)
-                    {
-                        Events.ErrorProcessingSubscriptions(e, identity);
-                    }
+                    Events.ProcessingSubscriptionsOnDeviceConnected(identity);
+                    await this.ProcessExistingSubscriptions(identity.Id);
                 }
+                catch (Exception e)
+                {
+                    Events.ErrorProcessingSubscriptions(e, identity);
+                }
+            }
+
+            try
```

Comment:
Should we add tests for this?

Reply:
I added a test for one of the tasks failing, but still making sure the other tasks finish